cpp/examples/src/example_server_flatbuffers.cpp

```diff
@@ -0,0 +1,141 @@
+static std::string getFileContents(std::string_view path) {
+  std::ifstream infile;
+  infile.open(path.data(), std::ios::binary | std::ios::in);
+  if (!infile) {
+    throw std::runtime_error("Could not open file " + std::string(path));
+  }
+  infile.seekg(0, std::ios::end);
+  int length = infile.tellg();
+  infile.seekg(0, std::ios::beg);
+  std::string result(length, '\0');
+  infile.read(result.data(), length);
+  infile.close();
+  return result;
+}
+
+int main(int argc, char** argv) {
+  std::vector<std::string> args(argv, argv + argc);
+  if (args.size() < 2) {
+    std::cerr << "Usage: example_server_flatbuffers /path/to/SceneUpdate.bfbs" << std::endl;
+    return 1;
+  }
+  const auto& sceneUpdateBfbsPath = args[1];
+
+  const auto logHandler = [](foxglove::WebSocketLogLevel, char const* msg) {
+    std::cout << msg << std::endl;
+  };
+  foxglove::ServerOptions serverOptions;
+  auto server = std::make_unique<foxglove::Server<foxglove::WebSocketNoTls>>(
+    "C++ FlatBuffers example server", logHandler, serverOptions);
+
+  foxglove::ServerHandlers<foxglove::ConnHandle> hdlrs;
+  hdlrs.subscribeHandler = [&](foxglove::ChannelId chanId, foxglove::ConnHandle) {
+    std::cout << "first client subscribed to " << chanId << std::endl;
+  };
+  hdlrs.unsubscribeHandler = [&](foxglove::ChannelId chanId, foxglove::ConnHandle) {
+    std::cout << "last client unsubscribed from " << chanId << std::endl;
+  };
+  server->setHandlers(std::move(hdlrs));
+  server->start("0.0.0.0", 8765);
+
+  const auto chanelIds = server->addChannels({{
+    .topic = "example_msg",
+    .encoding = "flatbuffer",
+    .schemaName = "foxglove.SceneUpdate",
+    .schema = Base64Encode(getFileContents(sceneUpdateBfbsPath)),
+  }});
+  const auto chanId = chanelIds.front();
+
+  bool running = true;
+
+  asio::signal_set signals(server->getEndpoint().get_io_service(), SIGINT);
+  signals.async_wait([&](std::error_code const& ec, int sig) {
+    if (ec) {
+      std::cerr << "signal error: " << ec.message() << std::endl;
+      return;
+    }
+    std::cerr << "received signal " << sig << ", shutting down" << std::endl;
+    running = false;
+  });
+
+  flatbuffers::FlatBufferBuilder builder;
+  builder.ForceDefaults(true);
+  while (running) {
+    builder.Clear();
+
+    const auto now = nanosecondsSinceEpoch();
+    auto timestamp = foxglove::Time(now / 1'000'000'000, now % 1'000'000'000);
+
+    auto pose = foxglove::CreatePose(
+      builder, foxglove::CreateVector3(builder, 2, 0, 0),
+      createQuaternionFromAxisAngle(builder, 0, 0, 1, double(now) / 1e9 * 0.5));
+    auto size = foxglove::CreateVector3(builder, 1, 1, 1);
+    auto color = foxglove::CreateColor(builder, 0.6, 0.2, 1, 1);
+    auto cubeBuilder = foxglove::CubePrimitiveBuilder(builder);
+    cubeBuilder.add_pose(pose);
+    cubeBuilder.add_size(size);
+    cubeBuilder.add_color(color);
+    const auto cube = cubeBuilder.Finish();
+    builder.Finish(cube);
```

Comment:
This call is unneeded (and may be harmful?). You should only call `builder.Finish` at the end.

Your problems were just that you had been building more than one flatbuffer table at once, when you are suppose to have called Finish (or called Create*, which calls Finish under the hood) before you start creating the new table.